library/core/class.session.php

```diff
@@ -26,6 +28,15 @@ class Gdn_Session {
     /** Maximum length of inactivity, in seconds, before a visit is considered new. */
     const VISIT_LENGTH = 1200; // 20 minutes
 
+    /** Array of ranked permissions. */
+    private const RANKED_PERMISSIONS = [
+        'Garden.Settings.Manage',
+        'Garden.Community.Manage',
+        'Garden.Moderation.Manage',
+        'Garden.Curation.Manage',
+        'Garden.SignIn.Allow',
+    ];
```

Comment:
It's possible that this will make your logic easier:

```suggestion
    private const RANKED_PERMISSIONS = [
        'Garden.Settings.Manage' => 1,
        'Garden.Community.Manage' => 2,
        'Garden.Moderation.Manage' => 3,
        'Garden.Curation.Manage' => 4,
        'Garden.SignIn.Allow' => 5,
    ];
```